File: blueprints/news/app/web/handler/response.go
package handler

import (
	"github.com/go-mizu/mizu"
)

// JSON response helpers

// Success returns a JSON success response.
func Success(c *mizu.Ctx, data any) error {
	return c.JSON(200, data)
}

// Created returns a JSON created response.
func Created(c *mizu.Ctx, data any) error {
	return c.JSON(201, data)
}

// NoContent returns a 204 response.
func NoContent(c *mizu.Ctx) error {
	return c.Text(204, "")
}

// BadRequest returns a 400 error.
func BadRequest(c *mizu.Ctx, message string) error {
	return c.JSON(400, map[string]string{"error": message})
}

// Unauthorized returns a 401 error.
func Unauthorized(c *mizu.Ctx) error {
	return c.JSON(401, map[string]string{"error": "unauthorized"})
}

// Forbidden returns a 403 error.
func Forbidden(c *mizu.Ctx) error {
	return c.JSON(403, map[string]string{"error": "forbidden"})
}

// NotFound returns a 404 error.
func NotFound(c *mizu.Ctx, resource string) error {
	return c.JSON(404, map[string]string{"error": resource + " not found"})
}

// Conflict returns a 409 error.
func Conflict(c *mizu.Ctx, message string) error {
	return c.JSON(409, map[string]string{"error": message})
}

// InternalError returns a 500 error.
func InternalError(c *mizu.Ctx, err error) error {
	return c.JSON(500, map[string]string{"error": "internal server error"})
}

File: blueprints/workspace/app/backend/hono/src/utils/response.ts
import type { Context } from 'hono';
import type { Env, Variables } from '../env';

export type AppContext = Context<{ Bindings: Env; Variables: Variables }>;

export function success<T>(c: AppContext, data: T, status: 200 | 201 = 200) {
  return c.json(data, status);
}

export function created<T>(c: AppContext, data: T) {
  return c.json(data, 201);
}

export function noContent(c: AppContext) {
  return c.body(null, 204);
}

export function badRequest(c: AppContext, message: string) {
  return c.json({ error: message }, 400);
}

export function unauthorized(c: AppContext, message = 'Unauthorized') {
  return c.json({ error: message }, 401);
}

export function forbidden(c: AppContext, message = 'Forbidden') {
  return c.json({ error: message }, 403);
}

export function notFound(c: AppContext, message = 'Not found') {
  return c.json({ error: message }, 404);
}

export function conflict(c: AppContext, message: string) {
  return c.json({ error: message }, 409);
}

export function tooManyRequests(c: AppContext, message = 'Too many requests') {
  return c.json({ error: message }, 429);
}

export function serverError(c: AppContext, message = 'Internal server error') {
  return c.json({ error: message }, 500);
}
